HW5/HW5 SQL.sql

```diff
@@ -0,0 +1,126 @@
+	SUM(past.off_walk) / NULLIF(SUM(past.off_strikeout), 0) AS off_walk_to_strikeout_avg,
+	SUM(past.def_walk) / NULLIF(SUM(past.def_strikeout), 0) AS def_walk_to_strikeout_avg,
+	SUM(past.off_ground) / NULLIF(SUM(past.off_flyout), 0) AS off_ground_to_flyout_avg,
+	SUM(past.def_ground) / NULLIF(SUM(past.def_flyout), 0) AS def_ground_to_flyout_avg,
+	SUM(past.off_hit) / NULLIF(SUM(past.off_at_bat), 0) AS off_batting_avg,
+	SUM(past.def_hit) / NULLIF(SUM(past.def_at_bat), 0) AS def_batting_avg,
+	SUM(past.off_plate_appr) / NULLIF(SUM(past.off_strikeout), 0) AS off_pa_so,
+	SUM(past.def_plate_appr) / NULLIF(SUM(past.def_strikeout), 0) AS def_pa_so
+FROM temp_for_rolling present
+JOIN temp_for_rolling past ON 
+	present.team_id = past.team_id
+	AND past.game_date BETWEEN DATE_SUB(present.game_date, INTERVAL 100 DAY) AND present.game_date
+	GROUP BY present.game_id, present.game_date, present.team_id
+;
+
+CREATE OR REPLACE TABLE temp_for_training
+SELECT
+	home.game_id AS game_id,
+	home.game_date AS game_date,
+	
+	home.team_id AS home_team_id,
+	home.avg_final_score AS home_final_score_avg,
+	home.off_batting_avg AS home_off_batting_avg,
+	home.def_batting_avg AS home_def_batting_avg,
+	home.off_pa_so AS home_off_plateapperance_strikeout,
+	home.def_pa_so AS home_def_plateapperance_strikeout,
+	home.avg_off_single AS home_off_single_avg,
+	home.avg_def_single AS home_def_single_avg,
+	home.avg_off_double AS home_off_double_avg,
+	home.avg_def_double AS home_def_double_avg,
+	home.avg_off_triple AS home_off_triple_avg,
+	home.avg_def_triple AS home_def_triple_avg,
+	home.off_walk_to_strikeout_avg AS home_off_walk_to_strikeout_avg,
+	home.def_walk_to_strikeout_avg AS home_def_walk_to_strikeout_avg,
+	home.off_ground_to_flyout_avg AS home_off_ground_to_flyout_avg,
+	home.def_ground_to_flyout_avg AS home_def_ground_to_flyout_avg,
+	
+	
+	away.team_id AS away_team_id,
+	away.avg_final_score AS away_final_score_avg,
+	away.off_batting_avg AS away_off_batting_avg,
+	away.def_batting_avg AS away_def_batting_avg,
+	away.off_pa_so AS away_off_plateapperance_strikeout,
+	away.def_pa_so AS away_def_plateapperance_strikeout,
+	away.avg_off_single AS away_off_single_avg,
+	away.avg_def_single AS away_def_single_avg,
+	away.avg_off_double AS away_off_double_avg,
+	away.avg_def_double AS away_def_double_avg,
+	away.avg_off_triple AS away_off_triple_avg,
+	away.avg_def_triple AS away_def_triple_avg,
+	away.off_walk_to_strikeout_avg AS away_off_walk_to_strikeout_avg,
+	away.def_walk_to_strikeout_avg AS away_def_walk_to_strikeout_avg,
+	away.off_ground_to_flyout_avg AS away_off_ground_to_flyout_avg,
+	away.def_ground_to_flyout_avg AS away_def_ground_to_flyout_avg,
+	
+	tbc.win AS home_team_win
+FROM 
+	temp_rolled home, 
+	temp_rolled away,
+	team_batting_counts tbc
+WHERE 1=1
+	AND home.team_id = tbc.team_id
+	AND away.team_id = tbc.opponent_team_id
+	AND tbc.homeTeam = 1
+	AND home.game_id = away.game_id
+	AND home.game_id = tbc.game_id
+;
+
```

Comment:
I know that in many cases WHERE and JOIN produce same results, but not sure if JOIN is more efficient than WHERE?

Reply:
I am not sure either, because it is easier for me to use.